crates/sui-core/src/authority/authority_store.rs

```diff
@@ -2133,14 +2271,41 @@ impl From<LockDetails> for LockDetailsWrapper {
 
 /// A potential input to a transaction.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct InputKey(pub ObjectID, pub Option<SequenceNumber>);
+pub enum InputKey {
+    Versioned {
```

Comment:
nit: should this be called `Object`?

Reply:
I like the suggestion of having `Object` in the name, and also think having `version` in there somewhere is useful as well. I'll rename to `VersionedObject`, but let me know if you have a preference for a different name.